CRM/Contribute/BAO/Contribution.php

```diff
@@ -3575,10 +3591,18 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
             $amount = $itemAmount;
             if (!$amount) {
               $amount = $params['total_amount'];
+              if ($context === NULL) {
+                $amount -= CRM_Utils_Array::value('tax_amount', $params, 0);
+              }
             }
           }
           else {
-            $amount = $diff * $fieldValues['line_total'];
+            if ($context == 'changedAmount') {
+              $itemParams['amount'] = $diff * ($fieldValues['line_total'] - $previousLineItem[$fieldValues['id']]['line_total']);
```

Comment:
also I think this whole logic :

```php
          if (!empty($params['is_quick_config'])) {
            $amount = $itemAmount;
            if (!$amount) {
              $amount = $params['total_amount'];
              if ($context === NULL) {
                $amount -= CRM_Utils_Array::value('tax_amount', $params, 0);
              }
            }
          }
          else {
            if ($context == 'changedAmount') {
              $itemParams['amount'] = $diff * ($fieldValues['line_total'] - $previousLineItem[$fieldValues['id']]['line_total']);
            }
            else {
              $amount = $diff * $fieldValues['line_total'];
            }
          }
```

could be moved to a new private method that return only the amount based on passed paramterts .. but it depndes on what this line $itemParams['amount'] = $diff * ($fieldValues['line'] ... etc   is for .